### PR TITLE
Convert the rest of System.Drawing printing to CsWin32

### PR DESCRIPTION
**Pull request by JeremyKuhne**
This moves the rest of System.Drawing.Printing to CsWin32.

- Move now-shared API to Core assembly
- Update calls to PInvokeCore for said calls
- Add HdcHandle for HDCs that we have to keep in fields
- Move printing DeviceContext usages to scopes
- Use new printer dialog to get default printer names
- Remove ScreenDC and use scope

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/10598)

### CODE REVIEW

**Comment by elachlan** [comment]
Does this resolve #10293 and #9879?

**Comment by JeremyKuhne** [comment]
> Does this resolve #10293 and #9879?

It only makes progress towards those

File: src/System.Drawing.Common/src/misc/DpiHelper.cs
Comment (DJm00n) [inline src/System.Drawing.Common/src/misc/DpiHelper.cs:1]: Maybe you can also remove `LogicalDpi` from `DpiHelper.cs` and `OneHundredPercentLogicalDpi` from `ScaleHelper.cs` and make use of `PInvoke.USER_DEFAULT_SCREEN_DPI`?

Comment (JeremyKuhne) [reply src/System.Drawing.Common/src/misc/DpiHelper.cs:1]: Thanks for pointing that out. I'll look at adding it.

File: src/System.Drawing.Common/src/System/Drawing/Graphics.cs [hunk 1]
```diff
@@ -3836,15 +3834,17 @@ private static void OnDomainUnload(object? sender, EventArgs e)
...
         // Generic error from GDI+ can be GenericError or Win32Error.
-        if (status == Gdip.GenericError || status == Gdip.Win32Error)
+        if (status is Gdip.GenericError or Gdip.Win32Error)
         {
             int error = Marshal.GetLastWin32Error();
             if (error == SafeNativeMethods.ERROR_ACCESS_DENIED || error == SafeNativeMethods.ERROR_PROC_NOT_FOUND ||
-                    // Here, we'll check to see if we are in a terminal services session...
-                    (((User32.GetSystemMetrics(NativeMethods.SM_REMOTESESSION) & 0x00000001) != 0) && (error == 0)))
+                    // Here, we'll check to see if we are in a terminal services session.
+                    (((PInvokeCore.GetSystemMetrics(SYSTEM_METRICS_INDEX.SM_REMOTESESSION) & 0x00000001) != 0) && (error == 0)))
```
Comment (lonitra) [inline src/System.Drawing.Common/src/System/Drawing/Graphics.cs:3847]: Is the 0x00000001 = `SYSTEM_METRICS_INDEX.SM_REMOTESESSION`? Maybe we can use that to be clear here.

File: src/System.Drawing.Common/src/misc/DpiHelper.cs [hunk 2]
```diff
@@ -157,7 +152,7 @@ public static bool IsScalingRequired
     /// <returns>The horizontal value in device units</returns>
     public static int LogicalToDeviceUnitsX(int value)
```
Comment (lonitra) [inline src/System.Drawing.Common/src/misc/DpiHelper.cs:153]: nit: body expression

File: src/System.Drawing.Common/src/misc/DpiHelper.cs [hunk 3]
```diff
@@ -169,7 +164,7 @@ public static int LogicalToDeviceUnitsX(int value)
     /// <returns>The vertical value in device units</returns>
     public static int ScaleToInitialSystemDpi(int value)
     {
-        return (int)Math.Round(LogicalToDeviceUnitsScalingFactorY * (double)value);
+        return (int)Math.Round(LogicalToDeviceUnitsScalingFactorY * value);
```
Comment (lonitra) [inline src/System.Drawing.Common/src/misc/DpiHelper.cs:167]: Ditto here.

**Comment by JeremyKuhne** [comment]
@lonitra I can address your feedback in a subsequent PR as I have a few in progress. :)

**Review by lonitra — APPROVED** [review]
:shipit:

**Pull request merged**